CONTRIBUTING.md

```diff
@@ -194,11 +161,48 @@ you can change the base branch using `git rebase --onto`, like this:
   $ git rebase --onto <NEW-BASE-BRANCH> <OLD-BASE-BRANCH> <YOUR-BRANCH>
 ```
 
-For example, backporting `my_branch` from `master` to `3.x`:
+For example, backporting `my_branch` from `master` to `4.x`:
+
+```bash
+  $ git rebase --onto 4.x master my_branch
+```
+
+### Making contributions that depend on pull requests in spyder-kernels
```

Comment:
@impact27, @dalthviz, @goanpeca, please review the instructions I added in this section to see if they are clear enough for you or can be improved.

This is going to be the new way to work with associated spyder-kernels pull requests from now on.

Reply:
> Maybe give a step-by-step example? something like:

Thanks a lot for this! I used it to improve the instructions I wrote.

> Maybe clarify if you need to do any of that if the PR touches only spyder-kernels

If you follow the guide, you'll see you only need to use git-subrepo here, because it's the repo where a subrepo is used.

> I am not sure why this is needed, won't both PR be merged at the same time?

Because if we merge the Spyder PR without cloning the repo again, it'll keep pointing to the user's clone and branch used in step 2. In other words, it's similar to reverting the changes to our installation scripts before merging in our current approach.

> I could e.g. modify the version in external-deps but not in my spyder-kernels folder, so the modifications would only be in the spyder repo?

You need to push first your modifications to your spyder-kernels branch, then pull them to your Spyder branch.